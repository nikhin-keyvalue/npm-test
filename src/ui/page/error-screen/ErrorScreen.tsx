import Preact, { h } from 'preact';

import { ErrorType, ZampfiPages } from 'constants/common';
import { ASSETS_PATH } from 'config';

import './style.scss';

interface ErrorScreenProps {
  onHandleButtonClick: (args: ZampfiPages) => void;
  errorType?: string;
}

const ErrorScreen: Preact.FunctionComponent<ErrorScreenProps> = ({
  onHandleButtonClick,
  errorType = ErrorType.TRANSACTION_FAILED
}) => (
  <div className='zamp-error-screen-wrapper'>
    {errorType === ErrorType.TRANSACTION_FAILED ? (
      <div className='error-icon-container'>
        <img src={`${ASSETS_PATH}/public/images/transaction-failed-icon.svg`} alt='transaction failed icon' />
        <div className='error-type-container'>Transaction Failed</div>
        <div className='error-message-container'>We are unable to allow this transaction at this point.</div>
      </div>
    ) : (
      <div className='error-icon-container'>
        <img src={`${ASSETS_PATH}/public/images/payment-failed-icon.svg`} alt='transaction failed icon' />
        <div className='error-type-container'>Payment Failed</div>
        <div className='error-message-container'>We are unable to proceed with this payment</div>
      </div>
    )}
    <button
      className='button-container'
      onClick={() => {
        onHandleButtonClick(ZampfiPages.WALLET_LIST);
      }}
    >
      CHOOSE ANOTHER WALLET
    </button>
  </div>
);

export default ErrorScreen;
