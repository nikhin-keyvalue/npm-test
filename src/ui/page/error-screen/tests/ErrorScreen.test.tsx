import { h } from 'preact';
import { configure, mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Adapter } from 'enzyme-adapter-preact-pure';

import ErrorScreen from '../ErrorScreen';

configure({ adapter: new Adapter() });

describe('Counter', () => {
  const onHandleButtonClickMock = jest.fn();

  it('render components', () => {
    const wrapper = mount(<ErrorScreen onHandleButtonClick={onHandleButtonClickMock} />);
    const tree = shallowToJson(wrapper);

    expect(tree).toMatchSnapshot();
  });

  it('handle back click', () => {
    const wrapper = shallow(<ErrorScreen onHandleButtonClick={onHandleButtonClickMock} />);

    wrapper.find('button.button-container').simulate('click');
    expect(onHandleButtonClickMock).toHaveBeenCalled();
  });
});
