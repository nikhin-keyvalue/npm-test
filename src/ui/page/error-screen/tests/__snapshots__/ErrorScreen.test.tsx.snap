// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter render components 1`] = `
<ErrorScreen
  onHandleButtonClick={[MockFunction]}
>
  <div
    className="zamp-error-screen-wrapper"
  >
    <div
      className="error-icon-container"
    >
      <img
        alt="transaction failed icon"
        src="undefined/public/images/transaction-failed-icon.svg"
      />
      <div
        className="error-type-container"
      >
        Transaction Failed
      </div>
      <div
        className="error-message-container"
      >
        We are unable to allow this transaction at this point.
      </div>
    </div>
    <button
      className="button-container"
      onClick={[Function]}
    >
      CHOOSE ANOTHER WALLET
    </button>
  </div>
</ErrorScreen>
`;
